mlibrary/my_account
# 2022-09-02 Updates

# Overview
This pull request relates to LIT's update-a-thon!

## NPM
`eslint` and `webpack` packages have been updated to their latest versions (as of September 2, 2022).

## Design System
### Spaces
The spacing variables provided by the Design System are being deprecated with the "glow up." All instances of `var(--space-...)` have been replaced by [their actual values](https://design-system.lib.umich.edu/design-tokens/), converted to `rem` measurements.

### `m-icon` Component
The `m-icon` component is also being deprecated with the Design System's "glow up." As an interim replacement, [Material Symbols](https://fonts.google.com/icons) has been imported and taken place where the components used to be.

**Note:** Some icons differ slightly between Material Symbols and the `m-icon` component. This is mostly noticeable in the `Account Overview` page.

#### Color Contrast Accessibility
With the icons now being text instead of images, contrast ratios are being checked against them, creating errors. Contrast ratios have been increased. The only exception is the `Favorites` card in Account Overview. This will be fine because the card will be going away at the end of the year, which will then remove the contrast ratio error.

## Sass
The Sass syntax has been updated since the site has launched. For instance, `@import` is being deprecated and recommends `@use` instead.

## Pull Request Template
In an effort to improve the development workflow, a pull request template has been added.

## Dockerfiles
Updated to conform to the current LIT convention

## Ruby Gems and Bundler
* Bundler updated to the latest version
* Gems updated to the latest version

## Testing
- Run the tests to make sure they pass (`docker-compose run --rm web bundle exec rspec`).
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Build the CSS (docker-compose run --rm web npm run build-css).
- Go through [localhost](http://localhost:4567/) and compare each page with the [live site](https://account.lib.umich.edu/). Make sure nothing is borked.
